packages/esm-patient-clinical-view-app/src/hooks/usePeers.ts

```diff
@@ -0,0 +1,112 @@
+  return person.attributes.reduce<{
+    contact: string | null;
+  }>(
+    (prev, attr) => {
+      if (attr.attributeType.uuid === config.contactPersonAttributesUuid.telephone) {
+        return { ...prev, contact: attr.display ? extractValue(attr.display) : null };
+      }
+      return prev;
+    },
+    {
+      contact: null,
+    },
+  );
+}
+
+function getPeer(relationship: Relationship, config: ConfigObject, person: 'personA' | 'personB') {
+  return {
+    ...extractAttributeData(relationship[person], config),
+    uuid: relationship.uuid,
+    name: extractName(relationship[person].display),
+    display: relationship[person].display,
+    relativeAge: relationship[person].age,
+    dead: relationship[person].dead,
+    causeOfDeath: relationship[person].causeOfDeath,
+    relativeUuid: relationship[person].uuid,
+    relationshipType: relationship.relationshipType.bIsToA,
+    patientUuid: relationship[person].uuid,
+    gender: relationship[person].gender,
+    startDate: !relationship.startDate ? null : formatDate(parseDate(relationship.startDate)),
+    age: relationship[person].age,
+    endDate: !relationship.endDate ? null : formatDate(parseDate(relationship.endDate)),
+  } as Peer;
+}
+
+function extractPeerData(
+  patientIdentifier: string,
+  relationships: Array<Relationship>,
+  config: ConfigObject,
+): Array<Peer> {
+  const relationshipsData: Peer[] = [];
+  let peerEducator: Peer | null = null;
+  for (const r of relationships) {
+    if (patientIdentifier === r.personA.uuid) {
+      relationshipsData.push(getPeer(r, config, 'personB'));
+    } else {
+      relationshipsData.push(getPeer(r, config, 'personA'));
+    }
+    if (peerEducator === null) {
+      if (patientIdentifier === r.personA.uuid) {
+        peerEducator = getPeer(r, config, 'personA');
+      } else {
+        peerEducator = getPeer(r, config, 'personB');
+      }
+    }
+  }
+  if (peerEducator) {
+    return [peerEducator, ...relationshipsData];
+  }
+  return relationshipsData;
+}
+
+const usePeers = (peerEducatorUuid: string) => {
+  const customeRepresentation =
+    'custom:(display,uuid,personA:(uuid,age,display,dead,causeOfDeath,gender,attributes:(uuid,display,value,attributeType:(uuid,display))),personB:(uuid,age,display,dead,causeOfDeath,gender,attributes:(uuid,display,value,attributeType:(uuid,display))),relationshipType:(uuid,display,description,aIsToB,bIsToA),startDate,endDate)';
+  const url = `/ws/rest/v1/relationship?v=${customeRepresentation}&person=${peerEducatorUuid}`;
+  const config = useConfig<ConfigObject>();
+  const { data, error, isLoading, isValidating } = useSWR<{ data: { results: Relationship[] } }, Error>(
+    url,
+    openmrsFetch,
+  );
+  const relationships = useMemo(() => {
+    return data?.data?.results?.length
+      ? extractPeerData(
+          peerEducatorUuid,
+          data?.data?.results.filter((rel) => rel.relationshipType.uuid === config.peerEducatorRelationship),
```

Comment:
Should we safe guard against undefined at runtime here

```suggestion
          data?.data?.results.filter((rel) => rel?.relationshipType?.uuid === config.peerEducatorRelationship),
```